### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by leogr**

### CODE REVIEW

File: schemas/core/data_type.go [hunk 1]
```diff
@@ -139,14 +139,15 @@ func NewDataType(t string) (DataType, error) {
 	case ComplexType:
 		return &Complex{}, nil
 	}
-	return nil, &dataTypeError{"Invalid type"}
+	return nil, &DataTypeError{"Invalid type"}
```
Comment (leogr) [inline schemas/core/data_type.go]: @alelb Message should be more descriptive (ie. including the invalid type name)

File: api/messages/error.go [hunk 2]
```diff
@@ -1,10 +1,51 @@
...
 type Error struct {
 	Schemas  []string `json:"schemas"`
-	Status   string   `json:"status"`
+	Status   string   `json:"status,required"`
 	ScimType string   `json:"scimType,omitempty"`
 	Detail   string   `json:"detail,omitempty"`
 }
+
+// ErrorWrapper wraps error in a scim Error struct
+func ErrorWrapper(e error) Error {
```
Comment (leogr) [inline api/messages/error.go]: It's a factory, `NewError` should be a better name.

**Review by leogr — COMMENTED** [review]
Almost OK!
Just minor fixes are needed prior to merge.

We need errors in filtering ASAP :)

File: api/messages/error_test.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+		Status:   string(http.StatusBadRequest),
+		ScimType: "invalidValue",
+		Detail:   err.Error(),
+	}, ErrorWrapper(err))
+
+	// Unmarshal type error
+	type Prova struct {
+		Num string `json:"num"`
+	}
+	p := Prova{}
```
Comment (leodido) [inline api/messages/error_test.go]: Hmmm ...

File: schemas/core/data_type.go (hunk 1 above)
Comment (leodido) [reply schemas/core/data_type.go]: Pass `t` handling message formatting within the `DataTypeError`.

Evaluate also to rename `DataTypeError`, into `InvalidDataTypeError`.

**Pull request merged**